File: src/main/java/com/langleon/dsobuildsim/dto/SetDTO.java
package com.langleon.dsobuildsim.dto;

import com.langleon.dsobuildsim.common.StatType;
import com.langleon.dsobuildsim.sets.SetType;

import java.util.Map;
import java.util.Set;

public record SetDTO(SetType setType, String name, int level, Set<String> setItems, Map<Integer, Map<StatType, Double>> baseValuesPerTier, Map<Integer, Map<StatType, Double>> relativeValuesPerTier, Map<Integer, String> descriptionPerTier) {
}

File: frontend/src/app/models/set-dto.ts
import { StatType } from "./stat-type";

export interface SetDTO {
    setType: string;
    name: string;
    level: number;
    setItems: string[];
    baseValuesPerTier: Record<string, Record<StatType, number>>;
    relativeValuesPerTier: Record<string, Record<StatType, number>>;
    descriptionPerTier: Record<string, string>;
}